src/viam/sdk/tests/test_proto_value.cpp

```diff
@@ -0,0 +1,366 @@
+#define BOOST_TEST_MODULE test module test_proto_value
+
+#include <memory>
+#include <unordered_map>
+
+#include <google/protobuf/struct.pb.h>
+
+#include <boost/mp11/algorithm.hpp>
+#include <boost/mp11/integer_sequence.hpp>
+#include <boost/mp11/tuple.hpp>
+#include <boost/test/included/unit_test.hpp>
+
+#include <viam/sdk/common/proto_value.hpp>
```

Comment:
It is a good habit to make the header under test the first include in the associated test, since it demonstrates that the header compiles in isolation. Usually we follow that practice in the `.cpp` for something as well, so it matters less for, say `proto_value.hpp` than it does for `proto_value_visit.hpp`, but generally just worth doing in my opinion.